packages/unuse-reactivity/src/unSignal/index.ts

```diff
@@ -102,10 +90,26 @@ export function unSignal<T>(initialValue?: T): UnSignal<T> {
 
       return value;
     },
-    peek: () => state.value,
-    set: setter,
-    update: (updater) => {
-      setter(updater(state.value));
+    peek() {
+      const prev = setCurrentSub(undefined);
+      const val = this.get();
+      setCurrentSub(prev);
```

Comment:
How is this different with previous solution?

Reply:
@JounQin do you have still open questions here, or could this be marked as solved?